### ISSUE TITLE
 Figwheel: Unable to dynam icly load figwheel.server.jetty-websocket/run-server

### ISSUE THREAD

**Issue by crowl** [bhauman/figwheel-main#49]
I'm getting this error after the cljs compilation succeds and the REPL starts. Any ideas?

```
...
[Rebel readline] Type :repl/help for online help info
Exception in thread "main" clojure.lang.ExceptionInfo: Figwheel: Unable to dynamicly load figwheel.server.jetty-websocket/run-server {:not-loaded figwheel.server.jetty-websocket/run-server}
        at clojure.core$ex_info.invokeStatic(core.clj:4739)
        at clojure.core$ex_info.invoke(core.clj:4739)
        at figwheel.repl$dynload.invokeStatic(repl.cljc:1070)
        at figwheel.repl$dynload.invoke(repl.cljc:1066)
        at figwheel.repl$run_default_server_STAR_.invokeStatic(repl.cljc:1089)
        at figwheel.repl$run_default_server_STAR_.invoke(repl.cljc:1086)
        at figwheel.repl$run_default_server.invokeStatic(repl.cljc:1123)
        at figwheel.repl$run_default_server.invoke(repl.cljc:1122)
        at figwheel.repl$setup.invokeStatic(repl.cljc:1152)
        at figwheel.repl$setup.invoke(repl.cljc:1147)
        at figwheel.repl.FigwheelReplEnv._setup(repl.cljc:1214)
        at cljs.repl$repl_STAR_$fn__6607.invoke(repl.cljc:942)
        at cljs.compiler$with_core_cljs.invokeStatic(compiler.cljc:1289)
        at cljs.compiler$with_core_cljs.invoke(compiler.cljc:1278)
        at cljs.repl$repl_STAR_.invokeStatic(repl.cljc:940)
        at cljs.repl$repl_STAR_.invoke(repl.cljc:855)
        at rebel_readline.cljs.repl$eval8558$repl_STAR___8559.invoke(repl.clj:77)
        at clojure.lang.Var.invoke(Var.java:385)
        at figwheel.main$repl.invokeStatic(main.cljc:1406)
        at figwheel.main$repl.invoke(main.cljc:1359)
        at figwheel.main$default_compile.invokeStatic(main.cljc:1614)
        at figwheel.main$default_compile.invoke(main.cljc:1579)
        at figwheel.main$build_main_opt.invokeStatic(main.cljc:495)
        at figwheel.main$build_main_opt.invoke(main.cljc:490)
        at cljs.cli$main.invokeStatic(cli.clj:636)
        at cljs.cli$main.doInvoke(cli.clj:625)
        at clojure.lang.RestFn.applyTo(RestFn.java:139)
        at clojure.core$apply.invokeStatic(core.clj:659)
        at clojure.core$apply.invoke(core.clj:652)
        at cljs.main$_main.invokeStatic(main.clj:61)
        at cljs.main$_main.doInvoke(main.clj:52)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:657)
        at clojure.core$apply.invoke(core.clj:652)
        at figwheel.main$_main$fn__7255.invoke(main.cljc:1908)
        at clojure.core$with_redefs_fn.invokeStatic(core.clj:7434)
        at clojure.core$with_redefs_fn.invoke(core.clj:7418)
        at figwheel.main$_main.invokeStatic(main.cljc:1906)
        at figwheel.main$_main.doInvoke(main.cljc:1885)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:702)
        at clojure.core$apply.invokeStatic(core.clj:657)
        at clojure.main$main_opt.invokeStatic(main.clj:317)
        at clojure.main$main_opt.invoke(main.clj:313)
        at clojure.main$main.invokeStatic(main.clj:424)
        at clojure.main$main.doInvoke(main.clj:387)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:702)
        at clojure.main.main(main.java:37)
```

This is my deps.edn file:

```
{:paths ["src" "resources"]
 :deps
 {org.clojure/clojure             {:mvn/version "1.9.0"}
  org.clojure/clojurescript       {:mvn/version "1.10.339"}
  org.clojure/core.async          {:mvn/version "0.4.474"}
  com.datomic/client-cloud        {:mvn/version "0.8.56"}
  com.cognitect/transit-clj       {:mvn/version "0.8.309"}
  com.cognitect/transit-cljs      {:mvn/version "0.8.256"}
  org.omcljs/om                   {:mvn/version "1.0.0-beta4"}
  io.pedestal/pedestal.service    {:mvn/version "0.5.4"}
  io.pedestal/pedestal.immutant   {:mvn/version "0.5.4"}
  com.stuartsierra/component      {:mvn/version "0.3.2"}}
  ch.qos.logback/logback-classic  {:mvn/version "1.2.3"
                                   :exclusions
                                   [org.slf4j/slf4j-api]}
  org.slf4j/slf4j-api             {:mvn/version "1.7.14"}
  org.slf4j/jul-to-slf4j          {:mvn/version "1.7.14"}
  org.slf4j/jcl-over-slf4j        {:mvn/version "1.7.14"}
  org.slf4j/log4j-over-slf4j      {:mvn/version "1.7.14"}}
 :aliases
 {:cljs.dev
  {:extra-paths ["target"]
   :extra-deps
   {com.bhauman/figwheel-main       {:mvn/version "0.1.5"}
    com.bhauman/rebel-readline-cljs {:mvn/version "0.1.4"}}}
  :cljs.admin.dev
  {:main-opts ["-m" "figwheel.main" "-b" "admin" "-r"]}
  :cljs.reserve.dev
  {:main-opts ["-m" "figwheel.main" "-b" "reserve" "-r"]}}}
```

**Comment by bhauman** [comment]
You will need to add exclusions to figwheel-main for `org.eclipse.jetty.websocket/websocket-servlet` `org.eclipse.jetty.websocket/websocket-server` `org.eclipse.jetty/jetty-server`

Then you will need to add dependencies for `org.eclipse.jetty.websocket/websocket-servlet` and `org.eclipse.jetty.websocket/websocket-servlet`  in the same version as the jetty server in Datomic client cloud.

**Comment by crowl** [comment]
Thanks, that dit it.

**Comment by martyglaubitz** [comment]
Im getting this error as well and i am using the project as generated by `lein new`

**Comment by bhauman** [comment]
I’m out and about Look in the docs under jetty conflicts

**Comment by bhauman** [comment]
Try using figwheel-main 0.2.0-SNAPSHOT it will provide a more detailed error in this case.

**Comment by rap1ds** [comment]
I had this exactly same issue.

After many trials and errors I figured out that in my case the conflicting dependency was caused by `[io.pedestal/pedestal.service 0.5.2]` which depends on `[ring/ring-core "1.5.1"]`. This conflicted with the ring version `[ring "1.6.3"]` that Figwheel 0.1.9 depends on. I fixed the issue by upgrading Pedestal to 0.5.5 which depends on the same ring version 1.6.3.
